src/site/pages/digital/src/containers/ExprToCircuitPopup/index.tsx

```diff
@@ -0,0 +1,217 @@
+                    <input title="Enter Circuit Expression" type="text"
+                               value={expression}
+                               placeholder="!a | (B^third)"
+                               onChange={e => setExpression({expression: e.target.value})} />
+                    <br/>
+
+                    <div className={"exprToCircuitPopupSettings"}>
+                        <div className={"exprToCircuitPopupNotation"}>
+                            <h3>Notation</h3>
+                            <input type="radio" id="|" name="format" checked={format === "|"} onChange={() => setFormat("|")} value="|" />
+                            <label htmlFor="|">Programming 1 (&amp;, |, ^, !)</label><br/>
+                            <input type="radio" id="||" name="format" onChange={() => setFormat("||")} value="||" />
+                            <label htmlFor="||">Programming 2 (&amp;&amp;, ||, ^, !)</label><br/>
+                            <input type="radio" id="+" name="format" onChange={() => setFormat("+")} value="+" />
+                            <label htmlFor="+">Algebraic 1 (*, +, ^, !)</label><br/>
+                            <input type="radio" id="+_" name="format" onChange={() => setFormat("+_")} value="+_" />
+                            <label htmlFor="+_">Algebraic 2 (*, +, ^, _)</label><br/>
+                            <input type="radio" id="OR" name="format" onChange={() => setFormat("OR")} value="OR" />
+                            <label htmlFor="OR">Literal 1 (AND, OR, XOR, NOT)</label><br/>
+                            <input type="radio" id="or" name="format" onChange={() => setFormat("or")} value="or" />
+                            <label htmlFor="or">Literal 2 (and, or, xor, not)</label><br/>
+                        </div>
+        
+                        <div className={"exprToCircuitPopupOptions"}>
+                            <h3>Options</h3>
+                            <input onChange={() => setIsIC(!isIC)} checked={isIC} type="checkbox" id="isIC" name="isIC" />
+                            <label htmlFor="isIC">Generate into IC</label>
+
+                            <br/>
+                            <br/>
+
+                            <label>Input Component Type:  </label>
+                            <select id="input"
+                                    value={input}
+                                    onChange={e => setInput(e.target.value)}
+                                    onBlur={e => setInput(e.target.value)}>
+                                <option key="Constant Low" value="Constant Low">Constant Low</option>
+                                <option key="Constant High" value="Constant High">Constant High</option>
+                                <option key="Button" value="Button">Button</option>
+                                <option key="Switch" value="Switch">Switch</option>
+                                <option key="Clock" value="Clock">Clock</option>
```

Comment:
I don't think you need a `key` here since it isn't in a loop, however according to my comment above, you might want to turn this into a loop but either way the key will be dynamically set